common/upload.js

```diff
@@ -44,6 +44,14 @@
 
                     }, 10);
 
+                    var fileExtensionFilter = scope.column.filenameExtFilter;
+                    // needs to be a comma separated list, i.e. ".jpg", ".png", ...
+                    scope.fileExtensions = "";
+                    if (fileExtensionFilter) {
+                        for (var j=0; j<fileExtensionFilter.length; j++) {
```

Comment:
I think you can simplify this logic by using the `join` function (`scope.column.filenameExtFilter.join(",")`).

Also we are doing this in two different places, in `form.controller.js` and  `upload.js`. We can modify the ermrestjs API to actually return a string of comma separated values instead (I'm not sure whether we should do that or not. That's just a suggestion).

Reply:
I just realized that the ermrestjs api is allowing string value for `filenameExtFilter`. We should change that to always return array. If it's a string, we should turn it into an array of length one.